### DISCUSSION TITLE
Linter with all rules from the alguidelines.dev

### DISCUSSION THREAD

**Discussion by Lazys** [microsoft/alguidelines#144] [Ideas]
Two ideas:
1. Linter with implemented all or most of the rules from tne alguidelines.dev - for the static code analysis right in the VS Code.
2. Service which can be plugged in to the Azure DevOps Pull Requests as the custom policy (similar topic https://github.com/StefanMaron/vsc-lintercop/issues/3). It could work like SonarQube integration with Azure DevOps: https://www.sonarqube.org/microsoft-azure-devops-integration/

Creazy ideas, but that would be awesome - sth what I know from the programming in the .Net :)

**Comment by waldo1001** [comment]
Yep - makes all the sense int he world!  There have been already some hints between @StefanMaron and me about it - nothing concrete yet though ;-)

**Comment by StefanMaron** [comment]
Absolutely agree! But use with care. I dont think it makes mich sense to analyze coding patterns with the linter. But the gudelines or best practices described here, can/should be implemented.

Also we should describe all the rules which are already covered by the linter in here.

But I must admit, I already have a hard time writing the code for the linter rule and fixing bugs and everything. So if anyone wants to help with documentation, I would be thankful. 🤓
